### PR TITLE
Feature/order mailer

### PR DESCRIPTION
**Pull request by TylerRockwell**
### Why?
https://github.com/smashingboxes/apprenticeship/tree/master/Curriculum/Backend/bookstore
Because user's like to have a receipt for their purchase

### What Changed?
Created a mailer that sends an invoice to the user when they place an order

### CODE REVIEW

File: features/step_definitions/purchase_steps.rb [hunk 1]
```diff
@@ -87,3 +87,9 @@
 Then(/^I am asked to review the order total$/) do
   expect(page).to have_content@book.price
 end
+
+Then(/^I am emailed an order invoice containing the books details, quantity, and order total$/) do
+  expect(last_email.body.encoded).to include(@book.title)
+  expect(last_email.body.encoded).to include(@book.price.to_s)
+  expect(last_email.body.encoded).to include("Total for this order:")
```
Comment (dacur) [inline features/step_definitions/purchase_steps.rb]: would you rather test that the actual total is there, i.e. `@book.total` ?

**Comment by dacur** [comment]
can't remember if i mentioned the `email_spec` gem before.  you don't need to change anything, just check it out when you have time.

https://github.com/email-spec/email-spec

:shipit:

File: features/step_definitions/purchase_steps.rb (hunk 1 above)
Comment (TylerRockwell) [inline features/step_definitions/purchase_steps.rb]: I changed it to check for the order total. I agree that's a better test.

**Comment by TylerRockwell** [comment]
I don't think you have. I'll look into it. Thanks for the review, @dacur !

**Pull request merged**